app/data_model/questionnaire_store.py

```diff
@@ -43,22 +43,21 @@ def set_metadata(self, to_set):
 
     def _deserialise(self, data):
         json_data = json.loads(data, use_decimal=True)
-        completed_blocks = [
-            Location.from_dict(location_dict=completed_block)
-            for completed_block in json_data.get('COMPLETED_BLOCKS', [])
-        ]
+        self.completed = Completed(
+            json_data.get('COMPLETED_BLOCKS', []),
+            json_data.get('COMPLETED_SECTIONS', []),
+        )
         self.set_metadata(json_data.get('METADATA', {}))
         self.answer_store = AnswerStore(json_data.get('ANSWERS'))
         self.list_store = ListStore.deserialise(json_data.get('LISTS'))
-        self.completed_blocks = completed_blocks
         self.collection_metadata = json_data.get('COLLECTION_METADATA', {})
 
     def serialise(self):
         data = {
             'METADATA': self._metadata,
             'ANSWERS': list(self.answer_store),
             'LISTS': self.list_store.serialise(),
-            'COMPLETED_BLOCKS': self.completed_blocks,
+            **self.completed.serialise(),
```

Comment:
Think its probably better that the completed object has its own key at this level, rather than use the properties attributes so we can make changes to the structure of it without affecting the top level of the questionnaire store.

    'COMPLETED': self.completed.serialise()